pkg/cloud/services/compute/instance.go

```diff
@@ -505,6 +515,28 @@ func (s *Service) getOrCreatePort(eventObject runtime.Object, clusterName string
 	}
 
 	record.Eventf(eventObject, "SuccessfulCreatePort", "Created port %s with id %s", port.Name, port.ID)
+	if portOpts.DisablePortSecurity == true || networkDisabledPortSecurity == true {
```

Comment:
Can't we create port with/without port security not updating after creating port?

Reply:
no, we cannot. 
At the port level, PortSecurity is immutable. 
For the network level, there is a bug reporter (https://github.com/kubernetes-sigs/cluster-api-provider-openstack/issues/913)

Reply:
I think it is possible from [gophercloud docs](https://github.com/gophercloud/gophercloud/blob/master/openstack/networking/v2/extensions/portsecurity/doc.go#L94-L120).

Reply:
Indeed. The link you provided is more efficient as it does not need retrieving the port to update it, I will update my PR accordingly.

Reply:
Thanks a lot